tests/runner/replicate/blockchain_1.json

```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "vname": "BLOCKNUMBER",
+    "type": "BNum",
+    "value": "100"
+  },
+  {
+      "vname" : "REPLICATE_CONTRACT",
```

Comment:
What does this entry do?

Reply:
blockchain JSON today is only for use in our testsuite. So it just contains a response which is the "new address" of a contract deployed from (i.e., origin being) the first address mentioned here.

It isn't easy to test all the scenarios you mention (including the failing ones) because we can't actually deploy a contract ourselves in our testsuite. i.e., the testsuite doesn't have a "state" to keep track of, unless we wholly extend the testsuite's state server to do all that.

I think these are scenarios that are better tested from the blockchain side. So perhaps @sandipbhoir can take a note to add you (and I as well) as a reviewer when this is implemented on the blockchain protocol side to ensure that it's fully tested.

Reply:
But blockchain.json is only used for input to the test, right? This means that the entry above is only used to determine the next deployment address. It doesn't test that a contract is actually being deployed there.